### PR TITLE
[ADD][supply_chain_settings] Generate Procurement using Call for Bids, TASK#3528, HU#329

### PR DESCRIPTION
**Pull request by hugho-ad**
This module adds a default purchase requisition on the wizards of
purchase and warehouses (Settings>Configuration>Werehouse | Purchase), to generate procurements using calls for bids

### CODE REVIEW

File: supply_chain_settings/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+{
+    "name": "supplier_invoice_number_unique",
```
Comment (zaoral) [inline supply_chain_settings/__openerp__.py]: @hugho-ad please add the correct name of the module.
A human readable name. Please Use "Supply Chain Settings"

File: supply_chain_settings/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "version": "1.0",
+    "author": "Vauxoo",
+    "category": "Accouting",
```
Comment (zaoral) [inline supply_chain_settings/__openerp__.py]: @hugho-ad please change category to "Settings"

File: supply_chain_settings/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    ],
+    "demo": [],
+    "data": [
+        "view/res_config_view.xml"
+    ],
+    "test": [],
+    "js": [],
+    "css": [],
+    "qweb": [],
+    "installable": True,
```
Comment (zaoral) [inline supply_chain_settings/__openerp__.py:45]: @hugho-ad please remove the auto_install and active keys. They are not used in v8

File: supply_chain_settings/models/purchase_requisition.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+#
+##############################################################################
+from openerp.osv import fields, osv
+
+
+class product_template(osv.osv):
+    _inherit = 'product.template'
+
+    _columns = {
+        'purchase_requisition': fields.boolean(
```
Comment (zaoral) [inline supply_chain_settings/models/purchase_requisition.py]: @hugho-ad Please delete this file. You dont need to overwrite the field purchase_requisition to add a default value. Instead, this value will be set by default to true or false using the configuration wizard you modificate.

File: supply_chain_settings/models/res_config.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+#
+##############################################################################
+from openerp import models, fields, api, _
+
+
+class purchase_config_settings(models.TransientModel):
+    _inherit = 'purchase.config.settings'
+
+    default_purchase_requisition = fields.Boolean(
+        _("Generate Procurement using Call for Bids"),
```
Comment (zaoral) [inline supply_chain_settings/models/res_config.py]: @hugho-ad the attributes inside a fields.x() definition are automactly translated. Please remove the _() calls into the fields definition

File: supply_chain_settings/models/res_config.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        products = self.env['product.template'].search([])
+        for product in products:
+            product.write({'purchase_requisition': True})
+
+
+class stock_config_settings(models.TransientModel):
+    _inherit = 'stock.config.settings'
+
+    default_purchase_requisition = fields.Boolean(
+        _("Generate Procurement using Call for Bids"),
```
Comment (zaoral) [inline supply_chain_settings/models/res_config.py]: @hugho-ad remove the _(). same as the comment in line 29.

**Comment by hugho-ad** [comment]
@zaoral , I've corrected all the observations

**Comment by zaoral** [comment]
@hugho-ad  Hi Hugo.
There are new reviews

NOTE: Please do not forget to add the task and user story number and link when you create the PR.

Task [vx#3528](https://www.vauxoo.com/#id=3528&view_type=form&model=project.task) HU [329](https://www.vauxoo.com/?debug=#id=329&view_type=form&model=user.story&action=726) Sprint 12

Please do this modifications:
- Delete the method `set_default_purchase_requistion` defined in the purchase and stock configuration wizard.
- You need to update your file `models/__init__.py` and remove the import of the `purchase_requisition` file because it would raise you a server error.
- Only for standard please change the folder name from `models` to `model`.
- Fix flake 8 warning `./models/res_config.py:22:1: F401 'api' imported but unused`
- This is a generic vauxoo module available in the odoo apps page. A community module. Please need to add the vauxoo icon to the module `static/description/icon.png`. You can extract the image for one of the others addons-vauxoo modules like `purchase_order_line_sequence`.
- In the translation https://github.com/Vauxoo/addons-vauxoo/pull/281/files#diff-cac7a23b73ed66011eeb8d4f51bf0347R22 you typep "ordenes de compras" to the translation of "procurement orders". Please make the correct translation.
- Add es.po translation.

**Comment by hugho-ad** [comment]
@zaoral , I've corrected all the new observations

**Comment by zaoral** [comment]
@hugho-ad excelent. +1.
@nhomar this PR is ready to review/merge

**Comment by zaoral** [comment]
This was PR was forgotten.
Was not merged in the properly time.

In order to merge, I will make a rebase and squash.
Then I will run a dummy PR for yoytec to check that everything is ok.

Best Regards.

**Comment by nhomar** [comment]
@zaoral can you point the dummy PR in order to merge it please?

**Pull request closed without merging**